com.unity.shadergraph/Editor/Data/Nodes/Input/Texture/ProceduralVirtualTextureNode.cs

```diff
@@ -21,8 +21,8 @@ public ProceduralVirtualTextureNode()
 
             vtProperty.displayName = "ProceduralVirtualTexture";
             vtProperty.overrideReferenceName = "MyPVT";
-            vtProperty.generatePropertyBlock = false;
             vtProperty.value.procedural = true;
+            vtProperty.value.shaderDeclaration = HLSLDeclaration.UnityPerMaterial;
```

Comment:
See [earlier conversation](https://github.com/Unity-Technologies/Graphics/pull/6176/files/63c3accd5db4a90400a363794e895796d3246b53#r739337016), this is still forcing _all_ nodes initially at UnityPerMaterial (both new and upgraded). 

- Should we have the overhead of version tracking if PVT is not really exposed yet (versus just "good communication")?
- Given that it's only used internally; do we prefer
  - "no project changes behavior"
  - or "all nodes start at the most common PerMaterial", potentially breaking projects, and also potentially fixing many projects where it wasn't clear that things were going subtly wrong ...

Reply:
I tend to lean towards "no project changes behavior" most of the time.  Only if we are near 100% sure we can tell what the user intended and we can fix the behavior for that intent, then we can do the unilateral behavior change.  In this case though, it's not clear if the users meant to use the Global path or the Material path.

What would happen if a user is using Global path and they upgrade to this change that sets their PVT to PerMaterial?

Reply:
I also feel like the communication overhead is larger than the versioning overhead in this case..  :)

Reply:
Though it is a bit weird in this case, as previously it kind of worked both for the Global path and the Material path (as long as it wasn't instanced), correct?  So we don't really know if the user intent was Global or PerMaterial.  We only know that Global rendered correctly, and PerMaterial rendered correctly sometimes.

If we default old PVT nodes to Global, would it continue to have this behavior?   Working in Global, and working in non-instanced Material?

Reply:
The actual code change for versioning is minimal:
`    public override int latestVersion => 1;
    public override void OnAfterMultiDeserialize(string json)
    {
        if (sgVersion == 0)
        {
            // version 0 was hard-coded to Global (kinda..)
            shaderDeclaration = HLSLDeclaration.Global;
            ChangeVersion(1);
        }
    }`

Reply:
Cool, the code change is indeed minimal :-) Have it running locally already, doesn't seem to need anything else to have existing nodes as `Global` after reimporting while indeed creating new nodes as `Per Material`.

Reply:
> Though it is a bit weird in this case, as previously it kind of worked both for the Global path and the Material path (as long as it wasn't instanced), correct? So we don't really know if the user intent was Global or PerMaterial. We only know that the Global path rendered correctly, and Material path rendered correctly when non-instanced.

We only know the user intent at runtime, based on their call to either `m_Stack.BindToMaterial(mat)` or to `m_Stack.BindGlobally()`. It indeed worked globally, but unless I'm confused, it never worked per-material _in an SRP-compatible way_, right (which was why it was changed to global in the first place)? So, yes, it was bound globally, and this indeed kinda worked for per-material use as well as long as it wasn't instanced ... But afaik that's a SRP-specific issue: I have a scene set up which does the exact same with Color properties instead of PVT, and setting `Shader.SetGlobalColor("MyColor", globalColor)` but then also setting `mat.SetColor("MyColor", localColor)` shows the same problem, all materials rendered in the same batch as `mat` will get `localColor`

Reply:
> If we default old PVT nodes to Global, would it continue to have this behavior? Working in Global, and working in non-instanced Material?

Yes, this is the screenshot of my set-up, and it is the same _before_ making shader changes, and _after_ making changes with your suggested addition to make sure the nodes get upgraded with shaderDeclaration set to Global
![PVT-Before](https://user-images.githubusercontent.com/49522280/143248209-1f24eef9-76bf-4378-989e-2e4af2ac7375.PNG)

On the left column, we should see red, green and blue, but because the first two rows are batch-rendered, they only get one color. The middle column is rendered as expected (a single global color). And the third column is also binding globally (to magenta) but then binding a local stack for the first quad (to yellow); again, because the first two quads are batch-rendered, they have the same color. **Note** that that third column is not fixed in this PR, as it's SRP-specific, not VT.

If we would decide not to version (so, upgraded nodes are also Per Material), my project upgrades to this:
![PVT-After-NoVersioning](https://user-images.githubusercontent.com/49522280/143248934-9fff173c-2a89-49fe-b641-7fd5dd628736.PNG)
The left column is fixed because instancing works properly now, but the globally bound cyan and magenta colors are missing. As said, upgrading as such would break all projects where `BindGlobally()` is used.

Reply:
> I also feel like the communication overhead is larger than the versioning overhead in this case.. :)

Weirdly enough, I'm starting to thing that communication would be even more important if we do versioning. If we don't do versioning, we'd have to communicate "_Hey, you'll see that `BindGlobally()` isn't working anymore, that's because you now have to explicitly set your nodes to `shaderDeclaration.Global` for this to work!_". And even if people didn't get the memo, their project will immediately show artifacts, they'll probably point to VT rather quickly (because they're binding a stack but it doesn't bind anymore) and we'd be able to help out.

If we do versioning, I'm not sure if we can do without proper communication? Even though the project didn't break, there are still bugs in there, so we'd still need to communicate the opposite message "_Hey, there were issues with `BindToMaterial()` specifically when rendered in a single SRP batch, even if your project looked fine; this is fixed for newly created PVT nodes, but for upgraded nodes you'll have to manually set your nodes to `shaderDeclaration.PerMaterial`_". If people missed the memo this time, it could potentially result in bug reports months later (depending on how soon the instancing issues show up, and whether or not devs will immediately think of VT instead of thinking "this is some kind of SRP/shader flickering issue")

Or maybe I'm overthinking this; the fact remains that PVT is not publicly available yet so there cannot be that many projects that we break, either way

Reply:
True.  There's not going to be a 100% solution I think.   If you think this approach is best, that works for me.